crates/cli/src/meta/mod.rs

```diff
@@ -407,6 +411,26 @@ pub async fn search_deployer(
     Ok(response_value)
 }
 
+/// checks if the given contract implements IDescribeByMetaV1 interface
+pub async fn implements_i_described_by_meta_v1(
+    client: &ReadableClientHttp,
+    contract_address: Address,
+) -> bool {
+    if !supports_erc165(client, contract_address).await {
+        return false;
+    }
+
+    let parameters = ReadContractParameters {
+        address: contract_address,
+        call: IERC165::supportsInterfaceCall {
+            interfaceID: get_interface_id(IDescribedByMetaV1::IDescribedByMetaV1Calls::SELECTORS)
```

Comment:
can `get_interface_id` be a method on the selectors?

Reply:
no, because selectors are alloy generated bindings from contract artifacts, so cant really inject such method into them unless we do a fork of alloy i guess

Reply:
@rouzwelt i think you can, i don't see why them being generated by a macro means you cant `impl` for them